### ISSUE TITLE
real root of (-1)^(2/3)

### ISSUE THREAD

**Issue by sashaalesin** [sympy/sympy#20678]
```python
>> eq = parse_expr('(-1) ** (2/3)')
(-1)**(2/3)
>> real_root(eq)
(-1)**(2/3)
>> real_root(eq).evalf()
-0.5 + 0.866025403784439*I
```
How i can get `1`?
[wolfram](https://www.wolframalpha.com/input/?i=%28-1%29%5E%282%2F3%29&assumption=%22%5E%22+-%3E+%22Real%22) can do this.

Thank you!

**Comment by naveensaigit** [comment]
Try this -
``` python
>>> eq = (-1)**Rational(1,3)
>>> real_root(eq)
-1
```

**Comment by sashaalesin** [comment]
@naveensaigit 

I know this, but I am getting data from the user as a string.

**Comment by Anoop01234** [comment]
I suppose wolfram is first calculating ((-1)**2))  =1 then calculating 1**(1/3) =1! So you may do it this way (((-1)**2)**(1/3)) which is actually equal to (-1)**(2/3), providing you the answer 1.0. This will give you the desired answer. Hope this is helpful

**Comment by oscarbenjamin** [comment]
I'm not sure what definition sympy is using for rational powers of a negative base.

If there is a clear definition being used then `real_root` could be extended to handle this case.

**Comment by sashaalesin** [comment]
@oscarbenjamin, hope it gets done! Thanks

**Comment by sylee957** [comment]
I think that the definition for rational power for complex base is ![image](https://user-images.githubusercontent.com/34944973/103241741-7a8ecb80-4997-11eb-8d91-8b78f92dc8ee.png)

**Comment by sylee957** [comment]
It comes from the identity in https://functions.wolfram.com/ElementaryFunctions/Power/04/02/05/ where ![image](https://user-images.githubusercontent.com/34944973/103243107-10c4f080-499c-11eb-933f-d8a2521308d0.png), so it follows from ![image](https://user-images.githubusercontent.com/34944973/103243221-8466fd80-499c-11eb-821f-d0c04bedac9f.png) when we consider integer exponents as special like how wolfram had used.
I'm not sure if such definition is natural, but it was likely the sympy standard before.

**Comment by oscarbenjamin** [comment]
The formula as given at wolfram has constraints:
```
(z**a)**b == z**(a*b),   -pi < im(a*log(z)) <= pi,  -1 < a <= 1,   b in Z
```
In this example `z=-1`, `a=2` and `b=3`. We have b an integer but neither of the other conditions is met.

**Comment by sylee957** [comment]
No, they are connected by or, so it should hold whenever either one is true.

**Comment by oscarbenjamin** [comment]
Oh, okay.

And actually I got that wrong because we have `b=1/3` in which case none of the conditions is met.

Okay so in
```
((-1) ** 2) ** (1/3)
```
we have `z=-1`, `a=2`, `b=1/3` and none of the conditions are met.

On the other hand given
```
((-1) ** (1/3)) ** 2
```
we have `z=-1`, `a=1/3`, `b=2` and then `b in Z` so the identity holds and
```
((-1)**(1/3))**2 = (-1)**(2/3)
```

But now we have to choose whether to take the real or principal root in `(-1)**(1/3)`:
```python
In [27]: r = roots(x**3 + 1)

In [28]: r
Out[28]: 
⎧       1   √3⋅ⅈ     1   √3⋅ⅈ   ⎫
⎨-1: 1, ─ - ────: 1, ─ + ────: 1⎬
⎩       2    2       2    2     ⎭

In [29]: [expand(ri**2) for ri in r]
Out[29]: 
⎡     1   √3⋅ⅈ    1   √3⋅ⅈ⎤
⎢1, - ─ - ────, - ─ + ────⎥
⎣     2    2      2    2  ⎦
```

It seems to me that this is precisely the job of the `real_root` function so it could be changed to give the real root in this case. The docstring says:
```
    Parameters
    ==========
    
    n : int or None, optional
        If *n* is ``None``, then all instances of
        ``(-n)**(1/odd)`` will be changed to ``-n**(1/odd)``.
        This will only create a real root of a principal root.
        The presence of other factors may cause the result to not be
        real.
```
Maybe that could be changed to handle cases of the form `(-n)**(m/odd)` with some possible conditions in `m`.

**Comment by sylee957** [comment]
In my opinion, real_root should be ambiguous unless `n` is specified explicitly.
If it is left optional, we should parse `1/n` for backward compatibility, but we won't need to parse `p/q` if `real_root(-1, 3) ** 2` or `real_root((-1)**2, 3)` can be used instead.

**Comment by oscarbenjamin** [comment]
So what do you suggest for getting the real root from `(-1)**(2/3)`?

**Comment by sylee957** [comment]
We should raise error for `(-1)**(2/3)` and may better deprecate automation of `n` for real_root

**Comment by oscarbenjamin** [comment]
That would seem unfortunate. When you are working in the reals then `(-1)**(2/3)` is uniquely defined since `(-1)**(1/3)==-1` so it doesn't matter whether you read it as `((-1)**2)**(1/3)` or `((-1)**(1/3))**2`. 

In general we work in the complex numbers and use the principal root but I thought the purpose of `real_root` was to work around that and get the expected result in the reals.

**Comment by sylee957** [comment]
When we allow numerators other than 1, we have other ambiguity like
![image](https://user-images.githubusercontent.com/34944973/103253630-47633100-49c5-11eb-8ac8-43c5430ac906.png), ![image](https://user-images.githubusercontent.com/34944973/103253644-5a760100-49c5-11eb-96bc-7435f82c0089.png) if the root is defined to choose the real branch, and the latter one cannot be comprehended like `(-1)**(1/3)` even though `(-1)**(2/6)` could be cancelled into it.

**Comment by oscarbenjamin** [comment]
If we allow introducing extra factors into the numerator and denominator like that then we get ambiguity even in the integer case e.g.:
```
-1 = (-1)**3 = (-1)**(6/2) == ((-1)**6)**(1/2) == 1**(1/2) == 1
```
Clearly we need to be careful in the symbolic case but I think if we have `z**(m/n)` with `m` even and `n` odd we can allow `real_root` to extract the minus sign.

**Comment by sylee957** [comment]
We need a solid logic to only handle the unary evaluation of roots if the exponent is rational.
But I think that `(-1)**(1/p)` is more easier to be extended for symbolic cases than `(-1)**(q/p)` because we have assumption for integers, but not for anything like even, odd or coprime.

**Comment by oscarbenjamin** [comment]
We have assumptions for even and odd:
```
In [1]: x = Symbol('x', even=True)

In [2]: x.is_even
Out[2]: True
```

**Comment by sashaalesin** [comment]
Why can't you find all the roots and then use the `real_root` to leave only the real ones?

In case `(-1) ** (1/3)` third root has 3 roots, of which 1 is real: `-1`. You can return it.
In case `(-1) ** (2/6)` sixth root has 6 roots, of which 2 are real: `-1` and `1`. You can return a tuple in this case.

[De Moivre's formula](https://en.wikipedia.org/wiki/De_Moivre%27s_formula) gives the roots in any case, you need leave only the real ones.

**Comment by smichr** [comment]
The symbolic expression (a Piecewise) from `real_root` does not -- cannot? -- give the correct result for arbitrary symbolic values of `n` (only the case when it is an integer):
```python
>>> from sympy.abc import x, y
>>> sol = real_root(x**y)
>>> [im(sol.subs(x,-1).subs(y,i)) for i in [-1,S.Half, 2/S(3), 2]]
[0, 1, sqrt(3)/2, 0]  <-- if there is a real root the imaginary part should be 0
```

**Comment by smichr** [comment]
> z**(m/n)

This can always be written as `(z**(1/n))**m` so we can return the real root of `z**(1/n)` raised to the power of `m`. But if the exponent is a symbolic and not something that for which `all(i.is_integer for i in expo.as_numer_denom())` is True then I don't think there is an expression that can be returned that will be correct once the known value is substituted -- unless we introduce functions to symbolically represent the numerator and denominator of a rational. So it would be best to raise an error in those cases.

**Comment by 80avin-sw** [comment]
I was able to solve this by patching `evalf_pow` function.
However, that works only for non-symbolic inputs like
```python
S('(-1)^(3/5)').evalf()
# -1
```

I also want it to work for symbolic inputs like
```python
S('(-x)^(3/5)').equals(S('((-x)^3)^(1/5)'))
# False
# should give True
```

Can anyone help me in this ?

**Comment by asmeurer** [comment]
The way the `real_root` function works is you provide the base as the first argument and the denominator of the exponent as the second argument, like `real_root(-1, 3)`. 

One way to make this work would be to note that `(-1)**Rational(2,3)` doesn't evaluate by default (if it did, you could pass `evaluate=False` to `sympify` or `parse_expr`), so you can replace powers with real_root using `replace`

```py
>>> sympify('(-1) ** (2/3)').replace(lambda expr: isinstance(expr, Pow) and isinstance(expr.exp, Rational), lambda expr: real_root(expr.base, expr.exp.q))
-1
```

Here the first argument of `replace` is a function that filters which subexpressions to replace, i.e., just powers with rational exponents, and the second argument is a function that gives the replacement.

Perhaps real_root should try to evaluate single expressions like `(-1)**(2/3)`. I'm not sure if that would lead to ambiguity in general, though. Either way, the above `replace` trick will work even for larger expressions containing `(-1)**(2/3)`.
